src/integrator_interface.jl

```diff
@@ -4,26 +4,53 @@
 Update solution of `integrator`, if necessary or forced by `force_save`.
 """
 function savevalues!(integrator::DDEIntegrator, force_save=false)
-    # update ODE integrator
-    update_ode_integrator!(integrator)
+    # update time of ODE integrator (can be slightly modified (< 10ϵ) because of time stops)
+    if integrator.integrator.t != integrator.t
+        if abs(integrator.t - integrator.integrator.t) >= 10eps(integrator.t)
+            error("unexpected time discrepancy detected")
+        end
+
+        integrator.integrator.t = integrator.t
+        integrator.integrator.dt = integrator.integrator.t - integrator.integrator.tprev
+    end
 
-    # update solution of ODE integrator
+    # update solution
     savevalues!(integrator.integrator, force_save)
 
-    if typeof(integrator.sol.prob) <: ConstantLagDDEProblem
-        #warn("ConstantLagDDEProblem is deprecated. Use DDEProblem instead.")
-        neutral = false
-        constant_lags = integrator.sol.prob.lags
-    else
-        neutral = integrator.sol.prob.neutral
-        constant_lags = integrator.sol.prob.constant_lags
+    # copy additional interpolation steps back from solution to reduced ODE integrator
```

Comment:
I am not quite sure why this is necessary?

Reply:
Because of line
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/integrators/integrator_utils.jl#L135
`integrator.integrator.k` is always reduced to a length of `integrator.kshortsize`.

Reply:
Maybe we should add a `reduce_size` argument to `savevalues!` of `ODEIntegrator`? This is cheap so it's fine, but it's not the nicest way to handle it.